lib/output/writer/elasticsearch.go

```diff
@@ -307,36 +309,49 @@ func (e *Elasticsearch) Write(msg types.Message) error {
 			return err
 		}
 
-		failed := result.Failed()
-		if len(failed) == 0 {
+		if !result.Errors {
+			requests = requests[:len(result.Items)]
```

Comment:
in case a subsequent loop has more requests to get through, I slice to skip them from being counted towards the offset using by `i` below